src/System.Security.Principal.Windows/tests/WindowsIdentityTests.Impersonate.cs

```diff
@@ -0,0 +1,709 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Security.Cryptography;
+using System.Security.Principal;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+// We skip all tests on Uap because we need to create some new Windows users before run
+// On Uap sandbox we cannot create users
+[SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+public class WindowsIdentityTestsImpersonate : IClassFixture<WindowsIdentityImpersonateFixture>
```

Comment:
These tests are modifying the current machine by adding user accounts.  They should all be outerloop.

Reply:
ok

Reply:
done

Reply:
/p:Outerloop=true

Reply:
https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.CoreFxTesting/build/core/test/Test.targets#L31

Reply:
strange command line it's ok(no -notrait category=OuterLoop)...but filter doesn't work
```
C:\git\corefx\src\System.Security.Principal.Windows (impersonate2 -> origin)                                                                                                                                                             
λ msbuild /t:rebuildandtest /v:m  /p:Outerloop=true                                                                                                                                                                                      
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework                                                                                                                                                                
Copyright (C) Microsoft Corporation. All rights reserved.                                                                                                                                                                                
                                                                                                                                                                                                                                         
  System.Security.Principal.Windows -> C:\git\corefx\artifacts\bin\ref\System.Security.Principal.Windows\netstandard-Debug\System.Security.Principal.Windows.dll                                                                         
  System.Security.Principal.Windows -> C:\git\corefx\artifacts\bin\System.Security.Principal.Windows\netcoreapp-Windows_NT-Debug\System.Security.Principal.Windows.dll                                                                   
  System.Security.Principal.Windows.Tests -> C:\git\corefx\artifacts\bin\System.Security.Principal.Windows.Tests\netstandard-Windows_NT-Debug\System.Security.Principal.Windows.Tests.dll                                                
  Using C:\git\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Debug-x64\ as the test runtime folder.                                                                                                                                
  Executing in C:\git\corefx\artifacts\bin\tests\System.Security.Principal.Windows.Tests\netcoreapp-Windows_NT-Debug-x64\                                                                                                                
  ----- start 17:31:35,20 ===============  To repro directly: =====================================================                                                                                                                      
  pushd C:\git\corefx\artifacts\bin\tests\System.Security.Principal.Windows.Tests\netcoreapp-Windows_NT-Debug-x64\                                                                                                                       
  set XUNIT_PERFORMANCE_MIN_ITERATION=1                                                                                                                                                                                                  
  set XUNIT_PERFORMANCE_MAX_ITERATION=1                                                                                                                                                                                                  
  set XUNIT_PERFORMANCE_MAX_ITERATION_INNER_SPECIFIED=1                                                                                                                                                                                  
  call C:\git\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Debug-x64\\dotnet.exe xunit.console.dll System.Security.Principal.Windows.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonwin
  tests  -notrait category=failing                                                                                                                                                                                                       
  popd                                                                                                                                                                                                                                   
  ===========================================================================================================                                                                                                                            
  xUnit.net Console Runner v2.4.1-pre.build.4059 (64-bit .NET Core 4.6.27130.03)                                                                                                                                                         
    Discovering: System.Security.Principal.Windows.Tests (method display = ClassAndMethod, method display options = None)                                                                                                                
    Discovered:  System.Security.Principal.Windows.Tests (found 16 test cases)                                                                                                                                                           
    Starting:    System.Security.Principal.Windows.Tests (parallel test collections = on, max threads = 8)                                                                                                                               
    Finished:    System.Security.Principal.Windows.Tests                                                                                                                                                                                 
  === TEST EXECUTION SUMMARY ===                                                                                                                                                                                                         
     System.Security.Principal.Windows.Tests  Total: 18, Errors: 0, Failed: 0, Skipped: 0, Time: 0,212s                                                                                                                                  
  ----- end 17:31:37,12 ----- exit code 0 ----------------------------------------------------------                                                                                                                                     
                                                                                                                                                                                                                                         
C:\git\corefx\src\System.Security.Principal.Windows (impersonate2 -> origin)                                                                                                                                                             
λ msbuild /t:rebuildandtest /v:m                                                                                                                                                                                                         
Microsoft (R) Build Engine version 15.9.21+g9802d43bc3 for .NET Framework                                                                                                                                                                
Copyright (C) Microsoft Corporation. All rights reserved.                                                                                                                                                                                
                                                                                                                                                                                                                                         
  System.Security.Principal.Windows -> C:\git\corefx\artifacts\bin\ref\System.Security.Principal.Windows\netstandard-Debug\System.Security.Principal.Windows.dll                                                                         
  System.Security.Principal.Windows -> C:\git\corefx\artifacts\bin\System.Security.Principal.Windows\netcoreapp-Windows_NT-Debug\System.Security.Principal.Windows.dll                                                                   
  System.Security.Principal.Windows.Tests -> C:\git\corefx\artifacts\bin\System.Security.Principal.Windows.Tests\netstandard-Windows_NT-Debug\System.Security.Principal.Windows.Tests.dll                                                
  Using C:\git\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Debug-x64\ as the test runtime folder.                                                                                                                                
  Executing in C:\git\corefx\artifacts\bin\tests\System.Security.Principal.Windows.Tests\netcoreapp-Windows_NT-Debug-x64\                                                                                                                
  ----- start 17:31:48,11 ===============  To repro directly: =====================================================                                                                                                                      
  pushd C:\git\corefx\artifacts\bin\tests\System.Security.Principal.Windows.Tests\netcoreapp-Windows_NT-Debug-x64\                                                                                                                       
  call C:\git\corefx\artifacts\bin\testhost\netcoreapp-Windows_NT-Debug-x64\\dotnet.exe xunit.console.dll System.Security.Principal.Windows.Tests.dll  -xml testResults.xml -notrait category=nonnetcoreapptests -notrait category=nonwin
  tests  -notrait category=OuterLoop -notrait category=failing                                                                                                                                                                           
  popd                                                                                                                                                                                                                                   
  ===========================================================================================================                                                                                                                            
  xUnit.net Console Runner v2.4.1-pre.build.4059 (64-bit .NET Core 4.6.27130.03)                                                                                                                                                         
    Discovering: System.Security.Principal.Windows.Tests (method display = ClassAndMethod, method display options = None)                                                                                                                
    Discovered:  System.Security.Principal.Windows.Tests (found 16 test cases)                                                                                                                                                           
    Starting:    System.Security.Principal.Windows.Tests (parallel test collections = on, max threads = 8)                                                                                                                               
    Finished:    System.Security.Principal.Windows.Tests                                                                                                                                                                                 
  === TEST EXECUTION SUMMARY ===                                                                                                                                                                                                         
     System.Security.Principal.Windows.Tests  Total: 18, Errors: 0, Failed: 0, Skipped: 0, Time: 0,207s                                                                                                                                  
  ----- end 17:31:49,95 ----- exit code 0 ----------------------------------------------------------                                                                                                                                     
                                                                                                                                                                                                                                         
C:\git\corefx\src\System.Security.Principal.Windows (impersonate2 -> origin)                                                                                                                                                             
λ                                                                                                                                                                                                                                        
```
Surely I missing something...however I use CI don't worry thank's for help!

Reply:
which filter do you mean? looking if we have a bug somewhere :)

Reply:
Also where is this coming from?

```
set XUNIT_PERFORMANCE_MIN_ITERATION=1                                                                                                                                                                                                  
  set XUNIT_PERFORMANCE_MAX_ITERATION=1                                                                                                                                                                                                  
  set XUNIT_PERFORMANCE_MAX_ITERATION_INNER_SPECIFIED=1         
```

I removed that months ago, I'm surprised these commands are in your logs.

Reply:
This branch is very old...maybe this is the issue. However I wanted run only Outerloop tests, I'll wait ongoing discussion and in case I'll rebase this branch to try.